actions/views/modals.ts

```diff
@@ -14,9 +14,15 @@ export function openModal<P>(modalData: ModalData<P>) {
     };
 }
 
+export type OpenModalReturnType = ReturnType<typeof openModal>
+
 export function closeModal(modalId: string) {
     return {
         type: ActionTypes.MODAL_CLOSE,
         modalId,
     };
 }
+
+export type CloseModalReturnType = ReturnType<typeof closeModal>
+
+export type ActionsReturnType = OpenModalReturnType | CloseModalReturnType;
```

Comment:
Replaced with `GenericAction` as we need to discuss this approach some more over at https://community.mattermost.com/core/pl/smb6hfdcuidu3rckja6y6p1gho